### PR TITLE
Create encounter: part 2

### LINKED ISSUE #448: Create encounter
Modal for creating an generic encounter

Image

Image

Image

### PR DESCRIPTION
**Pull request by mark-prins**
Closes #448 

Ability to add / edit an encounter. First steps anyway, will need to add the encounter page next.

### CODE REVIEW

File: client/packages/common/src/ui/forms/JsonForms/JsonForm.tsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ const FormComponent = ({
     user,
   };
 
-  return (
+  return !data ? null : (
```
Comment (mark-prins) [inline client/packages/common/src/ui/forms/JsonForms/JsonForm.tsx:97]: I was finding is misleading - the data was null and I was getting a spinner shown unceasingly

File: client/packages/system/src/Patient/Encounter/DetailModal.tsx [hunk 2]
```diff
@@ -1,95 +1,144 @@
...
   const patientId = usePatient.utils.id();
-
-  const { current, document, programType, reset, setProgramType } =
-    usePatientModalStore();
-
+  const t = useTranslation('patients');
+  const { current, reset, document } = usePatientModalStore();
   const [program, setProgram] = useState<ProgramRowFragmentWithId | null>(null);
-  const handleSave = useUpsertEncounter(
+  const [error, setError] = useState(false);
```
Comment (mark-prins) [inline client/packages/system/src/Patient/Encounter/DetailModal.tsx:25]: there's altogether too much state and hook calls for my liking, but I think all are required

File: client/packages/system/src/Patient/Encounter/DetailModal.tsx [hunk 3]
```diff
@@ -1,95 +1,144 @@
...
+  useEffect(() => {
+    if (data && data?.totalCount > 0) {
+      const documentRegistry = data.nodes[0];
+      if (documentRegistry) {
+        setCreateDoc({ data: {}, documentRegistry });
+        setError(false);
+        return;
+      }
+    } else {
+      setError(true);
```
Comment (mark-prins) [inline client/packages/system/src/Patient/Encounter/DetailModal.tsx:72]: the `placeholder program` encounter doesn't have a doc registry entry so I'm showing an error

File: client/packages/system/src/Patient/PatientView/AddButton.tsx [hunk 4]
```diff
@@ -36,6 +36,12 @@ export const AddButton = () => {
     SplitButtonOption<PatientModal>
   >(options[1] as SplitButtonOption<PatientModal>);
 
+  const onSelectOption = (option: SplitButtonOption<PatientModal>) => {
+    setSelectedOption(option);
```
Comment (mark-prins) [inline client/packages/system/src/Patient/PatientView/AddButton.tsx:40]: as requested.. I was finding the select and click action slightly irritating when testing.

Comment (clemens-msupply) [reply client/packages/system/src/Patient/PatientView/AddButton.tsx:40]: Yes feels good

**Review by clemens-msupply — APPROVED** [review]
Works nicely thanks!

**Pull request merged**